### ISSUE TITLE
Redefining Literal class variable in subclass has 'Incompatible type'

### ISSUE THREAD

**Issue by AugPro** [python/mypy#8796]
### Are you reporting a bug, or opening a feature request?
Bug
### Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.
```python
from typing import Literal

class A:
    foo : Literal['bar', 'spam']= 'bar'
    
class B(A):
    foo = 'spam'
```
### What is the actual behavior/output?
```
main.py:7: error: Incompatible types in assignment (expression has type "str", base class "A" defined the type as "Union[Literal['bar'], Literal['spam']]")
Found 1 error in 1 file (checked 1 source file)
```
### What is the behavior/output you expect?
I expected no error, as `B.foo` is still of type `Literal['bar', 'spam']`
### What are the versions of mypy and Python you are using?
mypy == 0.770
Python == 3.8
### What are the mypy flags you are using? (For example --strict-optional)
no flags

**Comment by msullivan** [comment]
Yup, that looks like a bug. Does it work as a workaround if you repeat the type annotation in the subclass?

**Comment by AugPro** [comment]
yes it does ! It also works if I overwrite foo in \_\_init\_\_ method for instance, and in that case no need to repeat type annotation

**Comment by peterschutt** [comment]
Is this the same issue?

```py
from typing import Generic, List, Literal, TypeVar

T = TypeVar("T", bound=str)


class Thing(Generic[T]):
    collection: List[T] 
    word: T


class SubThing(Thing[Literal["word"]]):
    collection = ["word"]
    word = "word"
```

```
test.py:14: error: Incompatible types in assignment (expression has type "str", base class "Thing" defined the type as "Literal['word']")
Found 1 error in 1 file (checked 1 source file)
```

Mypy seems happy with the `collection` override but not `word`.

py == 3.8.5 
mypy == 0.782

**Comment by embray** [comment]
Ditto with

```
$ mypy --version
mypy 0.812
```

and no additional flags.

**Comment by willfrey** [comment]
Just ran into this issue with 0.910.

**Comment by SewerynPiorkowski** [comment]
there's still the same problem 0.930

**Comment by amickael** [comment]
Is this fixed? Still running into this issue in `0.982`

**Comment by JelleZijlstra** [comment]
This was fixed in https://github.com/python/mypy/commit/57de8dbfaef646bf6f962377f5e1a6ec1b47889f which isn't in 0.982.

**Comment by amickael** [comment]
Ah makes sense, thank you!
